### PR TITLE
fix(ui): editable region boundaries not persisting on refresh

### LINKED ISSUE #55636: [bug]: moving editable region on refresh
1) Navigate to a multifile challenge: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/review-algorithmic-thinking-by-building-a-dice-game/step-9
2) Input some code over multiple lines:

```js
const updateScore = (selectedValue, achieved) => {
  totalScore += selectedValue.value;
  // some comment
}
```

3) Save code `Ctrl + S`
4) Refresh page

image

Seen in: https://forum.freecodecamp.org/t/user-editable-area-gets-smaller-after-refreshing-page/702480

---

I assume this has to do with the container defaulting to the size it is originally given (e.g. 1 line), even though the saved code spans more.

### PR DESCRIPTION
**Pull request by dwrik**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55636

**Issue**
Despite the challenge files being saved, only their contents (code) were being restored on a reload. This lead to incorrect editable region boundaries since it would fallback to the seed values.

**Fix**
Implemented restoration of editable region boundaries along with the file contents.

**Outcome**
Since we always restore the updated editable region boundaries from local storage, we always have the optimal bounds as per the saved code.

ScreenRecording2024-07-25at4 15 16PM-ezgif com-video-to-gif-converter

### CODE REVIEW

**Review by naomi-lgbt — APPROVED** [review]
This is clever!

**Comment by dwrik** [comment]
Hi @naomi-lgbt,

Thank you! Looking forward to getting this merged once it receives another approval.

Best,
dwrik

**Review by ojeytonwilliams — APPROVED** [review]
Nice and simple, I like it. LGTM :+1:

**Comment by dwrik** [comment]
@ojeytonwilliams

Thanks, glad you liked the approach 🙂

**Pull request merged**
